test/test_ops.py

```diff
@@ -0,0 +1,212 @@
+      SkipEntry('sinh', skip=reference_eager_int_skip),
+      SkipEntry('sort', skip=reference_eager_all_skip),
+      SkipEntry('sqrt', skip=reference_eager_int_skip),
+      SkipEntry('squeeze', skip=reference_eager_all_skip),
+      SkipEntry('std_mean', skip=reference_eager_float_skip),
+      SkipEntry('std', skip=reference_eager_float_skip),
+      SkipEntry('sum', skip=reference_eager_all_skip),
+      SkipEntry('svd', skip=reference_eager_float_skip),
+      SkipEntry('symeig', skip=reference_eager_float_skip),
+      SkipEntry('t', skip=reference_eager_all_skip),
+      SkipEntry('tan', skip=reference_eager_int_skip),
+      SkipEntry('tanh', skip=reference_eager_int_skip),
+      SkipEntry('tile', skip=reference_eager_all_skip),
+      SkipEntry('to_sparse', skip=reference_eager_all_skip),
+      SkipEntry('topk', skip=reference_eager_all_skip),
+      SkipEntry('transpose', skip=reference_eager_all_skip),
+      SkipEntry('var', skip=reference_eager_float_skip),
+  }
+
+  for op in op_db:
+    for op_name, variant_name, skip in skips:
+      if op.name == op_name and op.variant_test_name == variant_name:
+        if op.skips == ():
+          op.skips = (skip,)
+        else:
+          op.skips += (skip,)
+
+
+# Add skips to OpInfo (if any) for this test suite.
+update_skips()
+
+
+class TestOpInfo(TestCase):
+
+  def compare_with_eager_reference(self, torch_fn: Callable,
+                                   sample_input: SampleInput, **kwargs) -> None:
+
+    def cpu(sample: SampleInput):
+      # Similar to `numpy` method on SampleInput.
+      # Converts tensors to cpu tensors by calling .detach().cpu() on them
+      # Numbers, strings, and bool are preserved as is
+      # Lists, tuples and dicts are handled by calling this function recursively
+      def to_cpu(x):
+
+        def _cpu(t):
+          return t.detach().cpu()
+
+        if isinstance(x, torch.Tensor):
+          return _cpu(x)
+        elif isinstance(x, list):
+          return list(map(to_cpu, x))
+        elif isinstance(x, tuple):
+          return tuple(map(to_cpu, x))
+        elif isinstance(x, dict):
+          return {k: to_cpu(v) for k, v in x.items()}
+        elif isinstance(x, (numbers.Number, bool, str)):
+          return x
+
+        raise ValueError("Unknown type {0}!".format(type(x)))
+
+      cpu_sample_input, cpu_args, cpu_kwargs = to_cpu(sample.input), to_cpu(
+          sample.args), to_cpu(sample.kwargs)
+      return (cpu_sample_input, cpu_args, cpu_kwargs)
+
+    t_inp, t_args, t_kwargs = sample_input.input, sample_input.args, sample_input.kwargs
+    cpu_inp, cpu_args, cpu_kwargs = cpu(sample_input)
+
+    actual = torch_fn(t_inp, *t_args, **t_kwargs)
+    expected = torch_fn(cpu_inp, *cpu_args, **cpu_kwargs)
+
+    self.assertEqual(actual, expected, exact_dtype=True, exact_device=False)
+
+  @ops(op_db, allowed_dtypes=(torch.float32, torch.long))
+  def test_reference_eager(self, device, dtype, op):
+    if self.device_type != 'xla':
```

Comment:
Do you want to use `device` param instead? Just maintains the consistency throughout this function, since we don't use `self.device_type` in this method anywhere else.

Reply:
AFAIK, `device` corresponds the actual device with it's id eg. `cuda:0`, `cuda:1` while `device_type` gives us the device type like `cuda` or `xla`.

Reply:
You are correct. `self.device_type` is a neat shorthand. Another way to acquire the type is to instantiate an actual torch device from the string: `torch.device(device).type`.